# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerRegistrationInfo.h
# Commit: fb720435215f
# Full Hash: fb720435215fa5be3b2bd2101f3a54382954b70c
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-10-24 09:44:34
# Regressor Bug: 1544232
# File Overlap Count: 4
# Description:
#   Bug 1544232 - Limit lifetime extension of SWs by SWs to the sender's lifetime. r=edenchuang
#   
#   This patch introduces an explicit concept of lifetimes with mechanisms
#   in place so that actions taken by Clients (windows or non-ServiceWorker
#   orkers) will extend the lifetime of a ServiceWorker, but a ServiceWorker
# ==============================================================================

diff -r 25452f6623a0 -r fb720435215f dom/serviceworkers/ServiceWorkerRegistrationInfo.h
--- a/dom/serviceworkers/ServiceWorkerRegistrationInfo.h	Thu Oct 24 03:02:41 2024 +0000
+++ b/dom/serviceworkers/ServiceWorkerRegistrationInfo.h	Thu Oct 24 03:02:42 2024 +0000
@@ -11,6 +11,7 @@
 
 #include "mozilla/dom/IPCNavigationPreloadState.h"
 #include "mozilla/dom/ServiceWorkerInfo.h"
+#include "mozilla/dom/ServiceWorkerLifetimeExtension.h"
 #include "mozilla/dom/ServiceWorkerRegistrationBinding.h"
 #include "mozilla/dom/ServiceWorkerRegistrationDescriptor.h"
 #include "nsProxyRelease.h"
@@ -141,11 +142,14 @@
 
   bool IsCorrupt() const;
 
-  void TryToActivateAsync(TryToActivateCallback&& aCallback = nullptr);
+  void TryToActivateAsync(
+      const ServiceWorkerLifetimeExtension& aLifetimeExtension,
+      TryToActivateCallback&& aCallback = nullptr);
 
-  void TryToActivate(TryToActivateCallback&& aCallback);
+  void TryToActivate(ServiceWorkerLifetimeExtension&& aLifetimeExtension,
+                     TryToActivateCallback&& aCallback);
 
-  void Activate();
+  void Activate(const ServiceWorkerLifetimeExtension& aLifetimeExtension);
 
   void FinishActivate(bool aSuccess);
 
@@ -170,6 +174,8 @@
   ServiceWorkerInfo* GetByDescriptor(
       const ServiceWorkerDescriptor& aDescriptor) const;
 
+  ServiceWorkerInfo* GetByClientInfo(const ClientInfo& aClientInfo) const;
+
   // Set the given worker as the evaluating service worker.  The worker
   // state is not changed.
   void SetEvaluating(ServiceWorkerInfo* aServiceWorker);